protocols/mdns/tests/use-async-std.rs

```diff
@@ -126,3 +144,59 @@ async fn run_peer_expiration_test(config: Config) -> Result<(), Box<dyn Error>>
         }
     }
 }
+
+async fn run_no_expiration_on_close_test(config: Config) -> Result<(), Box<dyn Error>> {
+    let mut a = create_swarm(config.clone()).await?;
+    let mut b = create_swarm(config).await?;
+
+    let mut closed = false;
+    let mut dialed = false;
+
+    loop {
+        futures::select! {
+            ev = a.select_next_some() => match ev {
+                SwarmEvent::Behaviour(Event::Discovered(peers)) => {
+                    for (peer, addr) in peers {
+                        if peer == *b.local_peer_id() && !dialed {
+                            // Connect to all addresses of b to 'expire' all of them
+                            a.dial(addr)?;
+                            dialed = true;
+                        }
+                    }
+                }
```

Comment:
Would it be better to not iterate and just take the first one? Overriding the boolean state multiple times seems a bit odd.